src/components/DonationCardModal/DonationCardModal.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState } from 'react';
+import styles from './DonationCardModal.module.css';
+
+interface DonationCardModalProps {
+   title: string;
+   target: number;
+   amount: number;
+   onUpdateDonationAmount: (amount: number) => void;
+   onCloseModal: () => void;
+}
+
+function DonationCardModal({
+   title,
+   target,
+   amount,
+   onUpdateDonationAmount,
+   onCloseModal,
+}: DonationCardModalProps) {
+   const [donationAmount, setDonationAmount] = useState(0);
+
+   const handleBackgroundClick = (event: React.MouseEvent<HTMLDivElement>) => {
+      if (event.target === event.currentTarget) {
+         onCloseModal();
+      }
+   };
+
+   const handleAmountChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+      setDonationAmount(Number(event.target.value));
+   };
+
+   const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      onUpdateDonationAmount(donationAmount);
+      onCloseModal();
+   };
```

Comment:
Please please please read about this subject- [smart components and dumb components](https://medium.com/@thejasonfile/dumb-components-and-smart-components-e7b33a698d43)

This can be a good place to talk about this approach (that not everyone agree on) that has its pros and cons. Its good to know these terms :)